### PR TITLE
Add the "Recommended For You" anime recommendation algorithm

### PR DESCRIPTION
**Pull request by revam**
to be used on the dashboard.

### CODE REVIEW

**Comment by revam** [comment]
Forgot to say it depends on https://github.com/ShokoAnime/ShokoServer/pull/937 getting merged first.

**Comment by revam** [comment]
Need to change approval to be percent based instead of absolute values

File: Shoko.Server/API/v3/Controllers/SeriesController.cs [hunk 1]
```diff
@@ -182,6 +182,136 @@ public ActionResult<List<SeriesRelation>> GetAnidbRelationsBySeriesID([FromRoute
...
+                endDate = DateTime.Now;
+            if (endDate.HasValue && !startDate.HasValue)
+                return BadRequest("Missing start date.");
+            if (startDate.HasValue && endDate.HasValue)
+            {
+                if (endDate.Value > DateTime.Now)
+                    return BadRequest("End date cannot be set into the future.");
+                if (startDate.Value > endDate.Value)
+                    return BadRequest("Start date cannot be newer than the end date.");
+            }
```
Comment (Cazzar) [inline Shoko.Server/API/v3/Controllers/SeriesController.cs:221]: Are we able to do this validation via attributes, to avoid boilerplate code?

**Review by Cazzar — CHANGES_REQUESTED** [review]

File: Shoko.Server/API/v3/Controllers/SeriesController.cs (hunk 1 above)
Comment (revam) [reply Shoko.Server/API/v3/Controllers/SeriesController.cs:221]: Will check.

Comment (revam) [reply Shoko.Server/API/v3/Controllers/SeriesController.cs:221]: It seems like the `Range` attribute only supports static values, and there is no specific attribute for validating `DateTime` ranges, so I will prefer this boilerplate over the other solution of writing a custom attribute that would only be used once (resulting in even more boilerplate). And I still don't know if it loads the custom attribute into memory once the app starts (resulting in semi-static values) or if it checks the attribute every time the method is accessed.

Unless you (or any other members/users) know a better way then I will leave it there.

**Pull request merged**